src/main/java/org/prebid/server/bidder/dmx/DmxBidder.java

```diff
@@ -105,33 +98,37 @@ public DmxBidder(String endpointUrl, JacksonMapper mapper) {
         }
 
         final Site site = request.getSite();
-        final Publisher publisher = site.getPublisher();
-        if (site.getPublisher() != null) {
-            request.toBuilder()
-                    .site(site.toBuilder().publisher(publisher.toBuilder().id(updatedPublisherId).build()).build())
-                    .build();
-        } else {
-            request.toBuilder()
-                    .site(site.toBuilder().publisher(Publisher.builder().id(updatedPublisherId).build()).build())
-                    .build();
+        if (site != null) {
+            final Publisher publisher = site.getPublisher();
+            if (site.getPublisher() != null) {
+                request.toBuilder()
```

Comment:
you can do smth like this 
updatedPublisher = site.Publisher == null ? Publisher.builder().id(updatedPublisherId).build() : publisher.toBuilder().id(updatedPublisherId).build()